### PR TITLE
chore: migrate tests to playwright

### PR DESCRIPTION
**Pull request by ScrewTSW**
### What does this PR do?
* Removes vitest
* Switches tests to `pnpm`
* Updates Runner imports

### Screenshot / video of UI
https://github.com/containers/podman-desktop-extension-ai-lab/actions/runs/10738134201/job/29787897311

### What issues does this PR fix or reference?
#1640 

### How to test this PR?
N/A

### CODE REVIEW

**Review by axel7083 — COMMENTED** [review]
We probably need to wait for https://github.com/containers/podman-desktop-extension-ai-lab/pull/1653 to be merged as I don't think we should have a pnpm-lock.yaml in other places that the root of the workspace

File: tests/playwright/playwright.config.ts [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ***********************************************************************/
+
+import { defineConfig, devices } from '@playwright/test';
+
+export default defineConfig({
+  outputDir: 'tests/playwright/output/',
```
Comment (odockal) [inline tests/playwright/playwright.config.ts:22]: shouldn't it be relative path given the placement of the config? It applies for other locations in this file as well.

File: tests/playwright/package.json [hunk 2]
```diff
@@ -4,17 +4,18 @@
   "description": "Podman Desktop AI Lab extension Playwright E2E tests",
   "scripts": {
     "test:e2e:setup": "xvfb-maybe --auto-servernum --server-args='-screen 0 1280x960x24' --",
-    "test:e2e": "npm run test:e2e:setup vitest run src/ --pool=threads --poolOptions.threads.singleThread --poolOptions.threads.isolate --no-file-parallelism"
+    "test:e2e": "pnpm test:e2e:setup pnpm exec playwright test src/"
```
Comment (odockal) [inline tests/playwright/package.json:7]: no need to switch to pnpm here.

**Review by odockal — CHANGES_REQUESTED** [review]

**Review by benoitf — COMMENTED** [review]
There should be no pnpm lock file in tests/playwright directory being added

File: tests/playwright/playwright.config.ts (hunk 1 above)
Comment (ScrewTSW) [reply tests/playwright/playwright.config.ts:22]: I copied it 1 to 1 from bootc

File: tests/playwright/package.json [hunk 3]
```diff
@@ -4,17 +4,18 @@
...
   },
   "author": "Red Hat",
   "license": "Apache-2.0",
   "devDependencies": {
     "@playwright/test": "^1.46.1",
-    "@podman-desktop/tests-playwright": "1.12.0",
+    "@podman-desktop/tests-playwright": "^1.12.0",
```
[the remark this answers is not in the document]
Comment (ScrewTSW) [reply tests/playwright/package.json]: reverted

**Comment by ScrewTSW** [comment]
> There should be no pnpm lock file in tests/playwright directory being added

fixed

File: tests/playwright/package.json (hunk 2 above)
Comment (ScrewTSW) [reply tests/playwright/package.json]: reverted

File: tests/playwright/playwright.config.ts (hunk 1 above)
Comment (ScrewTSW) [reply tests/playwright/playwright.config.ts]: fixed.

**Comment by ScrewTSW** [comment]
> We probably need to wait for #1653 to be merged as I don't think we should have a pnpm-lock.yaml in other places that the root of the workspace

file removed

**Review by benoitf — APPROVED** [review]

**Review by odockal — APPROVED** [review]

**Pull request merged**